src/components/loginRegister/registerRequest.js

```diff
@@ -12,7 +12,10 @@ export const register = async (email, password) => {
     });
 
     if (!response.ok) {
-      throw new Error("Registration failed");
+      throw {
+        message: "Registration failed",
+        status: response.status,
+      };
```

Comment:
The stuff inside this block:
```
if (!response.ok) {
  // all request errors go into this block
}
```

Is where any possible error will go to. So you cant just assume that its an error because of a user already existing with that email. Theres tons of reasons why an error could occur in this request.

So if you're looking to throw an error because of an email already existing you'll need to check for that specific error.

The response body for this specific error will look like this:
```
{
  detail: 'account with that email already exists'
}

```
and it will have a status code of 400

So your code would maybe be something like this:

```
if (!response.ok) {
  const response_body = await response.json()
  if (response_body.detail.includes('email already exists')) {
    throw new Error('registration failed because a user with that email already exists')
  }
}
```

Reply:
Status codes are a general concept in web development and not just something that Im using. I wrote up a quick doc of how I use status codes in Propermesh:

https://github.com/rknizzle/propermesh/blob/master/docs/status_codes.md

So you can see that just checking a status_code of 400 isnt specific enough to know that its an error because of the email already existing.